src/init/features/project.ts

```diff
@@ -0,0 +1,133 @@
+import * as clc from "cli-color";
+import * as _ from "lodash";
+
+import { FirebaseProject, getProject, listProjects } from "../../firebaseApi";
+import * as logger from "../../logger";
+import { promptOnce, Question } from "../../prompt";
+import * as utils from "../../utils";
+
+const NO_PROJECT = "[don't setup a default project]";
+const NEW_PROJECT = "[create a new project]";
+
+/**
+ * Used in init flows to keep information about the project - basically
+ * a shorter version of FirebaseProject with some additional fields.
+ */
+export interface ProjectInfo {
+  id: string; // maps to projectId
+  label: string;
+  instance: string; // maps to FirebaseProject.resources.realtimeDatabaseInstance
+  location: string; // maps to FirebaseProject.resources.locationId
+}
+
+/**
+ * Get the user's desired project, prompting if necessary.
+ * Returns a ProjectInfo object:
```

Comment:
I'm fairly sure you can do this (which is a nicer way of describing the return type):

```suggestion
 * Returns a {@link ProjectInfo} object.
```

Spelling out the return type here is really redundant...

Reply:
Done